### PR TITLE
converted classcard and modulessidebar to tsx

### PR DESCRIPTION
**Pull request by dwu359**
Part of overall conversion to ts

### CODE REVIEW

**Comment by dwu359** [comment]
I guess I just happened to put the rest of the learning module tsx conversions here

**Comment by dwu359** [comment]
I also created an async thunk for fetching user progress data in the login store, but it's currently not being used rn since I'm not sure if the user progress data in the backend was set up correctly. So as of this commit, the learning modules do not get the user progress data

**Comment by dwu359** [comment]
ContentType is probably one of the craziest types I've written but oh well

File: backend/driver.py [hunk 1]
```diff
@@ -395,18 +395,20 @@ def writeToQueue() -> str:
 @app.route("/api/getUserProgressData", methods=["POST"])
 def getUserProgressData():
     dynamoTable = UserProgressDDBUtil(USERPROGRESS_TABLE_NAME, AWS_REGION)
+    user_id = json.loads(request.data)["user_id"]
+    print(user_id)
     try:
-        return dynamoTable.get_record(json.loads(request.data)).progressData
+        return dynamoTable.get_record(json.loads(request.data)["user_id"]).progressData
```
Comment (karkir0003) [inline backend/driver.py]: you already have a varaible called user_id here

File: backend/driver.py [hunk 2]
```diff
@@ -395,18 +395,20 @@ def writeToQueue() -> str:
...
     except ValueError:
-        newRecord = UserProgressData(json.loads(request.data), "{}")
+        newRecord = UserProgressData(json.loads(request.data)["user_id"], "{}")
```
Comment (karkir0003) [inline backend/driver.py]: same here

File: frontend/playground-frontend/src/components/AccountSettings/AccountSettings.tsx [hunk 3]
```diff
@@ -21,13 +18,17 @@ const SettingsBlock = () => {
       await updateUserSettings(fullName, email, password);
     }
   };
+
+  if (!user) {
+    return <></>;
```
Comment (karkir0003) [inline frontend/playground-frontend/src/components/AccountSettings/AccountSettings.tsx:23]: there is a bit of a cleaner way to do this. Have a helper function that renders the account settings form and do something like 

```
return (
{if user && renderAccountSettingsForm()}
)
```

File: frontend/playground-frontend/src/components/Home/Results.js [hunk 4]
```diff
@@ -277,24 +277,33 @@ const Results = (props) => {
...
           {problemType.value === "classification" &&
-          auc_roc_data_res.length !== 0 ? (
+          auc_roc_data_res.length !== 0 &&
+          props.simplified === false ? (
             <AUC_ROC_curves />
           ) : null}
           {problemType.value === "classification" &&
-          auc_roc_data_res.length === 0 ? (
+          auc_roc_data_res.length === 0 &&
+          props.simplified === false ? (
```
Comment (karkir0003) [inline frontend/playground-frontend/src/components/Home/Results.js]: what's `props.simplified`?

File: frontend/playground-frontend/src/components/Home/Results.js [hunk 5]
```diff
@@ -277,24 +277,33 @@ const Results = (props) => {
...
+            <ConfusionMatrix />
+          ) : null}
         </div>
       )}
     </>
   );
 };
 
+Results.defaultProps = {
```
Comment (karkir0003) [inline frontend/playground-frontend/src/components/Home/Results.js]: why is defaultProps here needed? @farisdurrani

File: frontend/playground-frontend/src/components/LearnMod/LearnMod.tsx [hunk 6]
```diff
@@ -0,0 +1,55 @@
+import React from "react";
+import ClassCard from "./ClassCard";
+import getContent from "./LearningModulesContent";
+import { useAppSelector } from "../../redux/hooks";
+//import { fetchUserProgressData } from "../../redux/userLogin";
```
Comment (karkir0003) [inline frontend/playground-frontend/src/components/LearnMod/LearnMod.tsx:5]: commented out code?

File: frontend/playground-frontend/src/components/LearnMod/LearnMod.tsx [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+
+const LearnMod = () => {
+  //module content
+  const content = getContent.modules;
+
+  const user = useAppSelector((state) => state.currentUser.user);
+  //const dispatch = useAppDispatch();
+  /*
+  useEffect(() => {
+    auth.onAuthStateChanged((user) => {
```
Comment (karkir0003) [inline frontend/playground-frontend/src/components/LearnMod/LearnMod.tsx:15]: commented out code?

**Review by karkir0003 — CHANGES_REQUESTED** [review]
address comments (some nitpicks)

File: frontend/playground-frontend/src/components/LearnMod/LearnMod.tsx (hunk 6 above)
Comment (dwu359) [reply frontend/playground-frontend/src/components/LearnMod/LearnMod.tsx:5]: I disabled fetching user progress data until Sebastian fixes the structure of the json for user progress data

File: frontend/playground-frontend/src/components/Home/Results.js (hunk 5 above)
Comment (farisdurrani) [reply frontend/playground-frontend/src/components/Home/Results.js]: We don't need this anymore. We want to remove all propTypes and initializing props outside of the arguments

File: frontend/playground-frontend/src/components/Home/Results.js [hunk 8]
```diff
@@ -277,24 +277,33 @@ const Results = (props) => {
...
+  simplified: false,
+};
+
 Results.propTypes = {
```
Comment (farisdurrani) [inline frontend/playground-frontend/src/components/Home/Results.js]: Delete propTypes

**Review by karkir0003 — APPROVED** [review]

**Review by farisdurrani — APPROVED** [review]

**Pull request merged**